### ISSUE TITLE
Removing or changing `x-camara-commonalities` extension field

### ISSUE THREAD

**Issue by rartych** [camaraproject/Commonalities#327]
**Problem description**
`x-camara-commonalities` field was introduced to [`info` object](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-design-guidelines.md#info-object) in version 0.4, but it occurs that it is hard to maintain esp. when Commonalities produce patch release. 

**Expected behavior**
Remove `x-camara-commonalities` field from `info` object definition in API Design Guidelines.
Release notes and Changelog include reference to the Commonalities release available when API is published.
Additionally, the same information can be mentioned in [API Readiness Checklist point 2](https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/API-Readiness-Checklist.md)

**Alternative solution**
Indicate only Commonalities release major number (r1) or version without patch (0.5).

**Additional context**

The issue was raised during [TSC meeting](https://lf-camaraproject.atlassian.net/wiki/spaces/CAM/pages/15434286/2024-10-03+TSC+Minutes#Commonalities-(%40Rafal-Artych-%40Pedro-D%C3%ADez-Garc%C3%ADa--)).

**Comment by murthygorty** [comment]
Agree, [changelog template](https://github.com/camaraproject/ReleaseManagement/blob/main/documentation/CHANGELOG_TEMPLATE.md) calls out the version of commonalities and ICM that the API is adhering to. 
x-camara-commonalities applied to a prior world when we didn't have current Release Management Standards.

BUT...
I request that we ask the 'Readiness Checklist' to reflect the commonalities version a little more strictly. The checklist guidelines require that "reference" column include the versions, but it doesnt look like our APIs are following it  - I checked against the first two API in our homepage, devicelocation and numberverfiication.

cc: @gmuratk

**Comment by murthygorty** [comment]
> @murthygorty The 'Readiness Checklist' was updated lately, what you have checked are probably checklists used in the previous release: https://github.com/camaraproject/ReleaseManagement/blob/r1.2/documentation/API-Readiness-Checklist.md
> 
> For Spring25 meta-release the new template shall be used and it will be validated by Release Management.

super, thanks @rartych
